src/client/user/components/CreateUrlModal/CreateLinkForm.tsx

```diff
@@ -300,6 +307,76 @@ const CreateLinkForm: FunctionComponent<CreateLinkFormProps> = ({
               </a>
             </CollapsibleMessage>
           </div>
+          <div className={classes.labelText}>
+            <Typography variant="body1">
+              Tag (add up to <strong>{MAX_NUM_TAGS_PER_LINK} tags</strong>){' '}
+              <Tooltip
+                title="Tags are words, or combinations of words, you can use to classify or describe your link."
+                imageAltText="Tags help"
+              />
+            </Typography>
+          </div>
+          <div>
+            <Autocomplete
+              multiple
+              options={['']}
+              value={tags}
+              PaperComponent={React.forwardRef(() => null)}
+              popupIcon={null}
+              closeIcon={null}
+              disabled={isUploading || tags.length >= MAX_NUM_TAGS_PER_LINK}
+              renderTags={(tags) =>
+                tags.map((tag) => (
+                  <FormTag
+                    key={tag}
+                    tag={tag}
+                    onClose={() => setTags(tags.filter((t) => t !== tag))}
+                  />
+                ))
+              }
+              renderInput={(params) => (
+                <TextField
```

Comment:
nit: Would it be better to abstract the `Autocomplete` component as another class. It might be cleaner to do so especially when it's not using any shared functions inside the `CreateLinkForm` component

Reply:
sounds good - i'll extract it as `TagAutocomplete`, and shift it to `client/user/widgets` so that it can be reused by the edit link drawer in the future as well!